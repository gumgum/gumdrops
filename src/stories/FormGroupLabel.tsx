import React, { ReactChild, ReactChildren } from 'react';
import cx from 'classnames';

export interface FormGroupLabelProps extends React.LabelHTMLAttributes<HTMLLabelElement> {
    text: ReactChildren | ReactChild;
    //htmlFor?: string;
}

export const FormGroupLabel: React.FC<FormGroupLabelProps> = ({
    text,
    className,
    ...otherProps
}) => (
    <label className={cx('gds-form-group__label', className)} {...otherProps}>
        {text}
    </label>
);
