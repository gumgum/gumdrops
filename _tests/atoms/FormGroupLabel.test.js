import renderer from 'react-test-renderer';
import React from 'react';
import FormGroupLabel from '../../components/atoms/FormGroupLabel';

test('Expect <FormGroupLabel> to render properly', () => {
    const defaultProps = {
        text: 'Label Baby Jr',
        className: 'my-classy-class'
    };
    const tree = renderer.create(<FormGroupLabel {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
});
