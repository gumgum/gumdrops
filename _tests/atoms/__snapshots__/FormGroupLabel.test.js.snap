// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <FormGroupLabel> to render properly 1`] = `
<label
  className="gds-form-group__label my-classy-class"
>
  Label Baby Jr
</label>
`;
